src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOpResolver.scala

```diff
@@ -32,11 +35,15 @@ private[marathon] class InstanceUpdateOpResolver(
   def resolve(op: InstanceUpdateOperation)(implicit ec: ExecutionContext): Future[InstanceUpdateEffect] = {
     op match {
       case op: Schedule =>
-        createInstance(op.instanceId) {
-          // TODO(karsten): Create events
+        // TODO(karsten): Create events
+        createInstance(op.instanceId){
           InstanceUpdateEffect.Update(op.instance, oldState = None, Seq.empty)
         }
-
+      case op: RelaunchReserved =>
```

Comment:
This is actually quite interesting. I think the logic in `InstanceUpdateOpResolver` is the one we need in the orchestrator 🙂

Reply:
yes, if orchestrator serializes updates per runspec and there is just one, we can have this in front of instance tracker, so in orchestrator :)

Reply:
I think that's 50% true – the resolver currently handles both updates to existing instances as well as forwarding the command to create them.

However the orchestrator would be the one that initially _decides_ to schedule an instance (based on e.g. an API request that says _please scale to 5_). It would however be the one that decides if e.g. a status update `TASK_FAILED` should affect the current goal (e.g. set new goal: Suspended if the orchestration logic has a retry limit).